### PR TITLE
ADD contract_partner module

### PR DESCRIPTION
**Pull request by angelmoya**
This module add service and contact partner in contract to use as addresses

### CODE REVIEW

File: contract_partner/README.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+================
+Contract Partner
+================
+
+This module was written to extend the functionality of contracts to support:
+* set contact and service partner in contract.
+* set contact and service partner from contract in invoice.
```
Comment (dreispt) [inline contract_partner/README.rst]: IMO the existing "Customer" field server as the "Contact", following v8  contact partner / commercial partner logic.

The "Service Partner" is the serviced address? How do you handle multi-site contracts? Do you feel necessary to push that information in the Invoice info?

Comment (angelmoya) [inline contract_partner/README.rst]: Yes @dreispt I can change partner term by address, really I use partner for addresses.

"Service Partner" is the serviced address, it's usefull on inovices, for example in pest control company they do services on restaurant franchise, or simply one company with more than one  warehose, for this company is usefull to get address on invoice.

"Contact Partner" maybe usefull in other repo, it's used to set address to send printed invoice.

Comment (dreispt) [inline contract_partner/README.rst]: The explanation for "contact" and "service" partner should be improved in the README:
- "contact" is the person or address where to send the invoices. If empty the "Customer" will be used. 
- "service" is where the service is rendered (site, location, address)

There is some overlapping with the [`service_desk`](https://github.com/OCA/project/tree/8.0/service_desk), addressing the same use cases but from the Project Task perspective:

The use case there is: 
1) I create a Task for a service
2) Pick a Contract (that has the `partner_id` "Customer" and an added `contact_id` Contact Partner)
3) Pick a Service Address. By default it's the Contact Partner but Customer child partners/addresses are selectable.

Looking at it, `contact_id` is in fact a mix between your contact and service fields. 
So I think that it should be deprecated to use the fields added here.

File: contract_partner/views/contract_view.xml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0"?>
+<openerp>
+    <data>
+
+        <record id="account_analytic_account_partner_form" model="ir.ui.view">
+            <field name="name">account.analytic.account.partner.form</field>
+            <field name="model">account.analytic.account</field>
+            <field name="inherit_id" ref="analytic.view_account_analytic_account_form" />
+            <field name="arch" type="xml">
+                <xpath expr="//field[@name='partner_id']" position="after">
```
Comment (dreispt) [inline contract_partner/views/contract_view.xml]: Why not use `<field name="partner_id" position="after">` ?

File: contract_partner/models/contract.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        string="Contact Address")
+
+    @api.model
+    def _prepare_invoice_data(self, contract):
+        invoice_vals = super(AccountAnalyticAccount, self).\
+            _prepare_invoice_data(contract)
+        if contract.service_partner_id:
+            invoice_vals['service_partner_id'] = contract.service_partner_id.id
+        if contract.contact_partner_id:
+            invoice_vals['contact_partner_id'] = contract.contact_partner_id.id
```
Comment (dreispt) [inline contract_partner/models/contract.py:23]: Don't Invoices already have a field with the person/address where the Invoice should be sent?

File: contract_partner/models/contract.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+class AccountAnalyticAccount(models.Model):
+    _inherit = 'account.analytic.account'
+
+    service_partner_id = fields.Many2one(
+        'res.partner',
+        string="Service Address")
+
+    contact_partner_id = fields.Many2one(
+        'res.partner',
+        string="Contact Address")
```
Comment (dreispt) [inline contract_partner/models/contract.py]: I would prefer just `Contact` - it could be a person.

File: contract_partner/models/contract.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+from openerp import models, fields, api
+
+
+class AccountAnalyticAccount(models.Model):
+    _inherit = 'account.analytic.account'
+
+    service_partner_id = fields.Many2one(
+        'res.partner',
+        string="Service Address")
```
Comment (dreispt) [inline contract_partner/models/contract.py]: I suggest `Service Location`. It's more generic: in the same address, I could have several locations, such as "Second Floor" or "Office 4B".

**Comment by rafaelbn** [comment]
Hi @angelmoya , could you please take a review here? It could be nice to have this merged. Thanks

**Comment by dreispt** [comment]
@angelmoya Let me know if any of my comments is not clear.

**Comment by angelmoya** [comment]
Sorry @dreispt and @rafaelbn I will review this week

File: contract_partner/models/contract.py (hunk 3 above)
Comment (angelmoya) [reply contract_partner/models/contract.py:23]: No, invoice have partner_id field, this partner_id is the address used to invoice, it is normal to invoice to one address and send to other address

File: contract_partner/models/contract.py (hunk 4 above)
Comment (angelmoya) [reply contract_partner/models/contract.py]: OK, I will change it

File: contract_partner/models/contract.py (hunk 5 above)
Comment (angelmoya) [reply contract_partner/models/contract.py]: OK, I will change it

**Comment by angelmoya** [comment]
I did it, @dreispt, is all ok? could you review?

**Comment by dreispt** [comment]
Did you push your code? I don't see any changes 😕

**Comment by angelmoya** [comment]
Sorry I did git stash for do only one commit, and i didn't push with -f, now it is updated

File: contract_partner/models/contract.py (hunk 3 above)
Comment (StefanRijnhart) [reply contract_partner/models/contract.py:23]: This is not in line with the Odoo data model. The related company is of course the parent company of the contact (partner_id) on the invoice. The service partner that you are adding looks quite idiosyncratic as well. As there is no real business logic, I'm thinking this change is not generic enough for the OCA.

Comment (pedrobaeza) [reply contract_partner/models/contract.py:23]: I agree.

**Review by rafaelbn — APPROVED** [review]
Just to say that this ""contact" is the person or address where to send the invoices. If empty the "Customer" will be used." is already done in v9 with "invoice address"

**Pull request closed without merging**